frontend/src/Metamaps/Topic.js

```diff
@@ -35,48 +36,41 @@ const Topic = {
     } else callback(DataModel.Topics.get(id))
   },
   launch: function(id) {
-    var start = function(data) {
-      Active.Topic = new DataModel.Topic(data.topic)
-      DataModel.Creators = new DataModel.MapperCollection(data.creators)
-      DataModel.Topics = new DataModel.TopicCollection([data.topic].concat(data.relatives))
-      DataModel.Synapses = new DataModel.SynapseCollection(data.synapses)
-      DataModel.attachCollectionEvents()
-
-      document.title = Active.Topic.get('name') + ' | Metamaps'
-
-      // set filter mapper H3 text
-      $('#filter_by_mapper h3').html('CREATORS')
-
-      // build and render the visualization
+    var start = function() {
       Visualize.type = 'RGraph'
       JIT.prepareVizData()
-
-      // update filters
-      Filter.reset()
-
-      // reset selected arrays
       Selected.reset()
-
-      // these three update the actual filter box with the right list items
+      Filter.reset()
       Filter.checkMetacodes()
       Filter.checkSynapses()
       Filter.checkMappers()
-
-      // for mobile
-      $('#header_content').html(Active.Topic.get('name'))
+      document.title = Active.Topic.get('name') + ' | Metamaps'
+      ReactApp.mobileTitle = Active.Topic.get('name')
+      ReactApp.render()
+    }
+    if (Active.Topic && Active.Topic.id === id) {
+      start()
```

Comment:
launch() and start() are kind of ambiguous names, can you make them more explicit/descriptive?